pkg/cluster/fixssh_test.go

```diff
@@ -149,16 +152,47 @@ func TestFixSSH(t *testing.T) {
 		}
 	}
 
+	ifNoVmBefore := func(lbID string, i int) *mgmtnetwork.Interface {
+		return &mgmtnetwork.Interface{
+			InterfacePropertiesFormat: &mgmtnetwork.InterfacePropertiesFormat{
+				VirtualMachine: nil,
+				IPConfigurations: &[]mgmtnetwork.InterfaceIPConfiguration{
+					{
+						InterfaceIPConfigurationPropertiesFormat: &mgmtnetwork.InterfaceIPConfigurationPropertiesFormat{
+							LoadBalancerBackendAddressPools: &[]mgmtnetwork.BackendAddressPool{
+								{
+									ID: to.StringPtr(fmt.Sprintf(lbID+"/backendAddressPools/ssh-%d", i)),
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	ifNoVmAfter := func(nic *mgmtnetwork.Interface) *mgmtnetwork.Interface {
+		emptyAddressPool := make([]mgmtnetwork.BackendAddressPool, 0)
+		(*nic.InterfacePropertiesFormat.IPConfigurations)[0].InterfaceIPConfigurationPropertiesFormat.LoadBalancerBackendAddressPools = &emptyAddressPool
+		return nic
+	}
```

Comment:
I know you are following the existing pattern, but this is a bad pattern. could we avoid having function variables in the test structs? (and in a follow up, fix the whole test file)

Reply:
I agree on the functions vs variables change. Let's make that change @mhsreddy and we can merge after that.

As an FYI, there is a large PUCM planned for Hive, so we'll need to be extra vigilant on merging this change.

Reply:
@facchettos I converted the variables to functions as you suggested. Please review the changes. I created a new Jira Story to refactor the entire test file.